lib/lcdbwf/tests/testthat/test-contrasts.R

```diff
@@ -0,0 +1,246 @@
+      }
+
+      # Check make_results output for each possible combination of test and type
+      expect_true(is_deseq_res(res$res))
+      expect_true(identical(names(res), c('res', 'dds', 'label')))
+      lrt_mcols_description <- paste0(as.character(lrt_design_data$design)[1], " ",
+                                      as.character(lrt_design_data$design)[2], "' vs '",
+                                      as.character(lrt_design_data$reduced_design)[1], " ",
+                                      as.character(lrt_design_data$reduced_design)[2], "'")
+      # 'Check' from the table above
+      # A
+      if ((is.null(test) || test == 'Wald') && (!is.null(type) && type == 'ashr')) {
+        expected_char <- paste(test %||% 'Wald', "test p-value:", contrast[1], contrast[2], "vs", contrast[3])
+        expect_true(mcols(res$res)$description[4] == expected_char)
+      # B
+      } else if ((is.null(test) || test == 'Wald') && (!is.null(type) && type == 'apeglm')) {
+        coef <- str_split(coef, "_")[[1]]
+        expected_char <- paste(test %||% 'Wald', "test p-value:", coef[1], coef[2], coef[3], coef[4])
+        expect_true(mcols(res$res)$description[4] == expected_char)
+      # C
+      } else if ((is.null(test) || test == 'Wald') && (is.null(type) || type == 'normal')) {
+        expected_char <- paste(test %||% 'Wald', "statistic:", contrast[1], contrast[2], "vs", contrast[3])
+        expect_true(mcols(res$res)$description[4] == expected_char)
+      # D
+      } else if ((!is.null(test) && test == 'LRT') && (!is.null(type) && type != 'normal')) {
+        expected_char <- paste0(test, " p-value: '", lrt_mcols_description)
+        expect_true(mcols(res$res)$description[4] == expected_char)
+      # E
+      } else if ((!is.null(test) && test == 'LRT') && (is.null(type) || type == 'normal')) {
+        expect_true(all(res$res$log2FoldChange == 0))
+        expected_char <- paste0(test, " statistic: '", lrt_mcols_description)
+        expect_true(mcols(res$res)$description[4] == expected_char)
+      } else {
+        stop(paste(label, 'was not checked'))
+      }
+      # Check for expected type stored in the result's metadata
+      if (!is.null(type)) {
+        expect_true(identical(metadata(res$res)$type, type))
+      } else if (is.null(type)) {
+        expect_true(is.null(metadata(res$res)$type))
+      }
+    }) # test_that
+  } # for type in shrinkage_types
+} # for test in tests
+
+
+# -------------------------- direct dds -------------------------- #
+test_that("make_results can handle dds object directly", {
+  dds <- dds_list[['dds_wald']]
+  # Directly pass the dds object
+  res <- make_results(dds_name=dds,
+                      label='Direct DDS',
+                      type='ashr',
+                      contrast=contrast)
+  expect_true(is_deseq_res(res$res))
+  expect_true(identical(names(res), c('res', 'dds', 'label')))
+  expect_true(any(grepl('Wald', mcols(res$res)$description[4])))
+}) # test_that
```

Comment:
Minor style comment: I don't think we need `# test_that` at the end of every single test. I think the current formatting/indentation makes it pretty clear, and it's not multiply nested so I feel like there's low confusion risk.